common/utils/yaml.go

```diff
@@ -0,0 +1,52 @@
+package utils
+
+import "gopkg.in/yaml.v3"
+
+func MergeYAML(oldValuesYamlContent string, newValuesYamlContent string) (string, error) {
+	if oldValuesYamlContent == "" {
+		return newValuesYamlContent, nil
+	}
+
+	if newValuesYamlContent == "" {
+		return oldValuesYamlContent, nil
+	}
+
+	var oldValuesYAML map[string]interface{}
+	var newValuesYAML map[string]interface{}
+
+	err := yaml.Unmarshal([]byte(oldValuesYamlContent), &oldValuesYamlContent)
+	if err != nil {
+		return "", err
+	}
+
+	err = yaml.Unmarshal([]byte(newValuesYamlContent), &newValuesYAML)
+
+	if err != nil {
+		return "", err
+	}
+
+	mergedValuesYAML := MergeMaps(oldValuesYAML, newValuesYAML)
+
+	mergedValues, err := yaml.Marshal(mergedValuesYAML)
+
+	return string(mergedValues), err
+}
+
+func MergeMaps(a, b map[string]interface{}) map[string]interface{} {
```

Comment:
Would this do the trick ? https://github.com/samber/lo?tab=readme-ov-file#assign

Reply:
I quickly read the function code and I think it would not. Because I need deep merge (my function recursively merge nested maps)